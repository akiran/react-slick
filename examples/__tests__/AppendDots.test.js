import React from "react";
import { mount } from "enzyme";
import AppendDots from "../AppendDots";
import { html as beautify_html } from "js-beautify";

function generateAppendDots(className){
  return dots => (
    <div
      className={className}
      style={{
        backgroundColor: "#ddd",
        borderRadius: "10px",
        padding: "10px"
      }}
    >
      <ul style={{ margin: "0px" }}>{dots}</ul>
    </div>
  );
}

describe("AppendDots Tests", () => {
  test("Default example test", () => {
    const slider = mount(<AppendDots />);
    const dotsContainer = slider.find("div.slick-dots");
    expect(dotsContainer.length).toBe(1);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Custom className test", () => {
    const settings = {
      appendDots: generateAppendDots('myClassName')
    };
    const slider = mount(<AppendDots settings={settings} />);
    const dotsContainer = slider.find("div.slick-dots.myClassName");
    expect(dotsContainer.length).toBe(1);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Custom className and dotsClass setting", () => {
    const settings = {
      appendDots: generateAppendDots('myClassName'),
      dotsClass: 'customClassName'
    };
    const slider = mount(<AppendDots settings={settings} />);
    const dotsContainer = slider.find("div.customClassName.myClassName");
    expect(dotsContainer.length).toBe(1);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  })
});