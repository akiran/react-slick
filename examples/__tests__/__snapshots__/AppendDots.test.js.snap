// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppendDots Tests Custom className and dotsClass setting 1`] = `
"<div>
    <h2>Append Dots</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <div class=\\"customClassName myClassName\\" style=\\"background-color: rgb(221, 221, 221); border-radius: 10px; padding: 10px;\\">
            <ul style=\\"margin: 0px;\\">
                <li class=\\"slick-active\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">1</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">2</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">3</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">4</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">5</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">6</div>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`AppendDots Tests Custom className test 1`] = `
"<div>
    <h2>Append Dots</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <div class=\\"slick-dots myClassName\\" style=\\"background-color: rgb(221, 221, 221); border-radius: 10px; padding: 10px;\\">
            <ul style=\\"margin: 0px;\\">
                <li class=\\"slick-active\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">1</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">2</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">3</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">4</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">5</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">6</div>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`AppendDots Tests Default example test 1`] = `
"<div>
    <h2>Append Dots</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <div style=\\"background-color: rgb(221, 221, 221); border-radius: 10px; padding: 10px;\\" class=\\"slick-dots\\">
            <ul style=\\"margin: 0px;\\">
                <li class=\\"slick-active\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">1</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">2</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">3</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">4</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">5</div>
                </li>
                <li class=\\"\\">
                    <div style=\\"width: 30px; color: blue; border: 1px solid blue;\\">6</div>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;
